bayes/inference_problem.py

```diff
@@ -81,10 +150,7 @@ def define_shared_latent_parameter_by_name(self, name):
                 self.latent[name].add(model_error.parameter_list, name)
 
     def loglike(self, number_vector):
-        self.latent.update(number_vector)
-        raw_me = {}
-        for key, me in self.model_errors.items():
-            raw_me[key] = me()
+        raw_me = self(number_vector)
```

Comment:
That is not part of the merge request, but I would suggest that we get rid of the tmp storage of the parameter list (both on the level of the model error and the inference problem), but generate these parameter lists on the fly, i.e. we have a method that takes as input the vector and returns the parameter_list. Alex might help here.